powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationServiceBehavior.java

```diff
@@ -595,11 +598,13 @@ public OperationDetailResponse cancelOperation(OperationCancelRequest request) t
                 .param("id", operationId)
                 .param("failureCount", operationEntity.getFailureCount())
                 .param("status", operationEntity.getStatus().name())
-                .param("additionalData", extendAuditedAdditionalData(operationEntity.getAdditionalData()))
+                .param("additionalData", extendAdditionalDataWithDevice(operationEntity.getAdditionalData()))
                 .build();
         audit.log(AuditLevel.INFO, "Operation canceled via explicit server call for operation ID: {}", auditDetail, operationId);
+        final OperationDetailResponse operationDetailResponse = convertFromEntity(savedEntity);
+        extendAdditionalDataWithDevice(operationDetailResponse);
```

Comment:
But correctly we are returning the user agent when canceling the operation. This is good.